run_task_request.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package tfe
+
+import (
+	"time"
+)
+
+const (
+	// RunTaskRegistrationRunId is the string that TFC/E sends when it just wants to register the Run Task.
+	RunTaskRegistrationRunId = "run-xxxxxxxxxxxxxxxx"
+)
```

Comment:
Would like to understand this more, along with the `IsEndpointValidation` check. Is the intent to ensure `r.RunID` is a valid run external ID (in the format of `RunTaskRegistrationRunID`)? As I read it, `IsEndpointValidation` will only return true if `r.RunID == "run-xxxxxxxxxxxx"` which should never be the case. 

If we do want to ensure that it's in proper external ID format, use the helper [validStringID()](https://github.com/hashicorp/go-tfe/blob/193d7a6b5066d8f4528ae0a38da39e7c82360761/validations.go#L29)